src/components/panel/panel.js

```diff
@@ -1636,6 +1685,91 @@ MdPanelRef.prototype._animateClose = function() {
   });
 };
 
+/**
+ * Registers a listener with the panel. The callback should return a promise,
+ * which will allow the action to continue when it gets resolved, or will
+ * prevent an action if it is rejected.
+ * @param {string} type Type of listener.
+ * @param {!angular.$q.Promise<!any>} callback Callback to be registered.
+ * @returns {!MdPanelRef}
+ */
+MdPanelRef.prototype.registerListener = function(type, callback) {
```

Comment:
I think these methods should be moved to a `/utils/CallbackRegistry.js` for generic usages.

```
this.callbacks = new CallbackRegistry(this);

this.callbacks.add(type, callback);
this.callbacks.remove(type, callback);

this.callbacks.notify(type);
this.callbacks.removeAll(type);

```